index.bs

```diff
@@ -816,62 +868,111 @@ To <dfn>handle funky elements</dfn> on a given |element|, run these steps:
     1. Remove the `formaction` attribute from |element|.
 </div>
 
-### The Effective Configuration ### {#configuration}
+## Matching Against The Configuration ## {#configuration}
 
-A Sanitizer is potentially complex, so we will define a helper
-construct, the *effective configuration*. This is mostly a specification
-convenience and allows us to explain a Sanitizer's operation in two steps:
-One, how to derive the effective configuration, and two, define the
-Sanitzer's operation based on it.
+A <dfn>sanitize action</dfn> can have the values `keep`, `drop`, or `block`.
 
-An effective configuration maps a given |element| or a given pair of
-|element| and |attribute| to a [=sanitize action=].
+<div algorithm="sanitize action for an element">
+To determine the <dfn>sanitize action for an |element|</dfn>, given a
+Sanitizer configuration dictionary |config|, run these steps:
 
-A <dfn>sanitize action</dfn> can have the values `keep`, `drop`, or `block`.
-To determine the <dfn>stricter action</dfn> of two [=sanitize actions=], pick
-the 'larger' of the two actions assuming a transitively defined order with
-`drop` &gt; `block`, and `block` &gt; `keep`.
+  1. Let |kind| be |element|'s [=element kind=].
+  1. If |element| is of `regular` |kind| and does not
+     [=element matches an element name|match=] any name in the
+     [=baseline element allow list=]: Return `drop`.
+  1. If |element| is of `custom` |kind| and if |config|'s
+     [=allow custom elements option=] is unset or set to anything other
+     than `true`: Return `drop`.
+  1. If |element| [=element matches an element name|matches=] any name
+     in |config|'s [=element drop list=]: Return `drop`.
+  1. If |element| [=element matches an element name|matches=] any name
+     in |config|'s [=element block list=]: Return `block`.
+  1. If |config| has a non-empty [=element allow list=] and
```

Comment:
Perhaps you could move the "empty" check to be part of the matching algorithm?

Reply:
Sort of done. (See below.)